Backpack/src/main/java/net/skyscanner/backpack/calendar2/adapter/CalendarAdapter.kt

```diff
@@ -19,47 +19,66 @@
 package net.skyscanner.backpack.calendar2.adapter
 
 import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import net.skyscanner.backpack.calendar2.CalendarFooterAdapter
+import net.skyscanner.backpack.calendar2.DefaultCalendarFooterAdapter
 import net.skyscanner.backpack.calendar2.data.CalendarDay
 import net.skyscanner.backpack.calendar2.data.CalendarFooter
 import net.skyscanner.backpack.calendar2.data.CalendarHeader
 import net.skyscanner.backpack.calendar2.data.CalendarItem
 import net.skyscanner.backpack.calendar2.data.CalendarMonth
 import net.skyscanner.backpack.calendar2.data.CalendarSpace
-import net.skyscanner.backpack.calendar2.extension.getItemByGlobalIndex
+import net.skyscanner.backpack.calendar2.extension.cellByPosition
+import net.skyscanner.backpack.calendar2.extension.cellsCount
 import net.skyscanner.backpack.calendar2.extension.yearMonthHash
 import net.skyscanner.backpack.util.Consumer
 import net.skyscanner.backpack.util.ItemHolder
 import org.threeten.bp.temporal.ChronoField
 
 internal class CalendarAdapter(
-  private val footers: CalendarFooterAdapter<*>,
+  private val scope: CoroutineScope,
   private val output: Consumer<CalendarDay>,
 ) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), Consumer<List<CalendarMonth>> {
 
   private var data: List<CalendarMonth> = emptyList()
 
+  var footerAdapter: CalendarFooterAdapter<*> = DefaultCalendarFooterAdapter
+    set(value) {
+      field = value
+      notifyDataSetChanged()
+    }
+
   init {
     setHasStableIds(true)
```

Comment:
i think you might not need stable ids + overwriting the `getItemId` method anymore if you're using diffutil

Reply:
TBH I see no harm in that and assume that this still may improve the performance in some cases